### PR TITLE
Add HTTP get request to fetch new api data and use it in JSX elements

### PR DESCRIPTION
**Pull request by christiemolloy**
- Adds the ApiServices file with the getApis function.
- Updates the hub base class file so that the httpGet method makes an Axios request instead of an Angular HTTP call.
- Updates the config service file so that the hubURL points to the new url: https://studio-api.apicur.io/  rather than local host
- Updates the App data list item so that it makes the call to the server (we should probably make this call from a different component I was just testing it here, any thoughts @dlabaj  ?)
- the Api data that is returned back is stored in the shared state, using the setState hook.
- the data list item imports apiData from the store Context state and this data is mapped over the data list item JSX elements

### CODE REVIEW

File: packages/services/src/api-services/hub.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        return options;
+    }
+
+    /**
+     * Converted to React/Axios
+     * Performs an HTTP GET operation to the given URL with the given options.  Returns
+     * a Promise to the HTTP response data.
+     */
+    protected httpGet<T>(url: string, options: any, successCallback?: (value: T) => T): Promise<any> {
+      options["observe"] = "response"; // not sure what this does?
```
Comment (EricWittmann) [inline packages/services/src/api-services/hub.ts]: This is a feature of the Angular HTTP client.  You can find more info on it here if interested:  https://angular.io/guide/http#reading-the-full-response

Comment (EricWittmann) [reply packages/services/src/api-services/hub.ts]: Spoiler alert - you probably don't need it with Axios.

File: packages/services/src/config/config.service.ts [hunk 2]
```diff
@@ -22,8 +22,8 @@ import { ConfigType } from './config.type';
 const DEFAULT_CONFIG: ConfigType = 
   {
     apis: {
-      editingUrl: "wss://studio-ws.apicur.io/api-editing",
-      hubUrl: "https://studio.apicur.io/api-hub",
+      editingUrl: "https://studio-api.apicur.io/api-editing",
```
Comment (EricWittmann) [inline packages/services/src/config/config.service.ts]: This should be:

wss://studio-ws.apicur.io

Comment (EricWittmann) [reply packages/services/src/config/config.service.ts]: That URL is only used by the editor - it makes a web socket connection when editing an API.

File: packages/services/src/api-services/hub.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  /**
+   * Creates a hub API endpoint from the api path and params.
+   * @param path
+   * @param params
+   * 
+   */
+  protected endpoint(path: string, params?: any, queryParams?: any): string {
+      if (params) {
+          for (let key in params) {
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: Change to (**const** key in params)

File: packages/services/src/api-services/hub.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+              let value: string = encodeURIComponent(params[key]);
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: This can be const instead of let as well.

File: packages/services/src/api-services/hub.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+              path = path.replace(":" + key, value);
+          }
+      }
+      let rval: string = this.apiBaseHref + path;
+      if (queryParams) {
+          let first: boolean = true;
+          for (let key in queryParams) {
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: Can be const instead of let

File: packages/services/src/api-services/hub.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+              if (queryParams[key]) {
+                  let value: string = encodeURIComponent(queryParams[key]);
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: Can be const instead of let.

File: packages/services/src/api-services/hub.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                  if (first) {
+                      rval = rval + "?" + key;
+                  } else {
+                      rval = rval + "&" + key;
+                  }
+                  if (value != null && value != undefined) {
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: !==

File: packages/services/src/api-services/hub.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+  /**
+   * Creates an editing endpoint from the given relative path and params.
+   * @param path
+   * @param params
+   * 
+   */
+  protected editingEndpoint(path: string, params?: any): string {
+      if (params) {
+          for (let key in params) {
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: const

File: packages/services/src/api-services/hub.ts [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+              let value: string = encodeURIComponent(params[key]);
```
Comment (dlabaj) [inline packages/services/src/api-services/hub.ts]: const

File: packages/services/src/api-services/index.ts [hunk 10]
```diff
@@ -0,0 +1,3 @@
+
+export * from './api-services';
+export * from './hub';
```
Comment (dlabaj) [inline packages/services/src/api-services/index.ts]: do we need to export hub?

File: packages/services/src/config/config.service.ts [hunk 11]
```diff
@@ -18,24 +18,44 @@
...
+    features: {
+        microcks: true
+    },
+    mode: "dev",
+    ui: {
+        url: "http://localhost:8888/"
+    },
+    user: undefined,
 };
+// const DEFAULT_CONFIG: ConfigType = {
```
Comment (dlabaj) [inline packages/services/src/config/config.service.ts]: Maybe add a comment that this is used for local development with a locally running server.

**Pull request merged**